### PR TITLE
Mark password properties sensitive

### LINKED ISSUE #13: Password properties should be sensitive
Let's leverage the `sensitive` flag on properties that has been introduced with Chef 12.14.34

### PR DESCRIPTION
**Pull request by olive42**
Per-property sensitivity was added in Chef 12.14, so make a check for it.

Fix #13

### CODE REVIEW

File: resources/api.rb [hunk 1]
```diff
@@ -3,7 +3,11 @@
 property :args, [Hash, String, NilClass], desired_state: false
 property :endpoint, String, desired_state: false, identity: true, default: 'http://localhost:8081/service/siesta/rest/v1/script/'.freeze
 property :username, String, desired_state: false, identity: true, default: 'admin'.freeze
-property :password, String, desired_state: false, identity: true, default: 'admin123'.freeze
+if ::Gem::Requirement.new('>= 12.14.34').satisfied_by?(::Gem::Version.new(::Chef::VERSION))
```
Comment (Annih) [inline resources/api.rb]: OPT: you can increase the chef version requirement in metadata.rb and support only chef >= 12.14.34

Comment (olive42) [inline resources/api.rb]: Is this ok to impose a newer version on users of the cookbook?

Comment (Annih) [inline resources/api.rb]: If they want to stay with the older version, you don't impose anything :P

Comment (olive42) [inline resources/api.rb]: Done.

Comment (Annih) [inline resources/api.rb]: So please remove the if :D

**Pull request merged**